tests/test_app.py

```diff
@@ -0,0 +1,82 @@
+from page_analyzer.utils.url import check_url, normalize, validate
+
+
+def test_validate():
+    # given
+    url = "https://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    expected_errors = {}
+    assert errors == expected_errors
+
+
+def test_validate_invalid():
+    # given
+    url = "htttps://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    expected_errors = {"name": "Некорректный URL"}
+    assert errors == expected_errors
+
+
+def test_normalize():
+    # given
+    url = "http://www.example.com:1030/software/index.html"
+
+    # when
+    normalized_url = normalize(url)
+
+    # then
+    expected_url = "http://www.example.com"
+    assert normalized_url == expected_url
+
+
+def test_check_url():
+    # given
+    url = "https://krisha.kz/"
+
+    # when
+    check_result = check_url(url)
+
+    # then
+    expected_result = {
```

Comment:
not necessary, but it'd be cool to see tests for `parse_page_data` function as well